### DISCUSSION TITLE
Custom Field - Select / choice - with right management

### DISCUSSION THREAD

**Discussion by lsgIT-HH** [kimai/kimai#3226] [Questions & Answers]
Hello, 

I am searching for a solution to add a custom field to the timesheet to choose / select a given entry from a list.
The special feature here is that the respective user should only be able to see entries assigned to him.

Is there any possibility to do this?
I already added a custom editable textfield using a MetaFieldSubscriber, but I do not know how to add a choicelist and also to filter this corresponding the assignment.

Would it be possible to achieve this using the custom fields plugin?

**Comment by kevinpapst** [comment]
This is a too special use-case for the plugin. You can do it with your custom code.
Here is an example of a custom field: 
https://github.com/Keleo/DemoBundle/blob/master/EventSubscriber/MetaFieldSubscriber.php#L37

The backend of this is Symfony and Symfony forms. So if you don't know how to use that, you should start reading their docs first I guess: https://symfony.com/doc/current/reference/forms/types/choice.html

If you want to display something from Kimai, you might want to use the EntityType in combination with a Repository query instead.

**Reply by lsgIT-HH** [reply]
Thanks for the answer. Unfortunately, I have not yet managed as desired.
Is it somehow possible to add custom javascript into my own plugin?

**Reply by kevinpapst** [reply]
https://www.kimai.org/documentation/developers.html#adding-custom-meta-tags-stylesheets-or-javascript
